napari/layers/base/base.py

```diff
@@ -130,17 +131,20 @@ def __init__(
         self.scale_factor = 1
 
         self.dims = Dims(ndim)
+
         if scale is None:
-            self._scale = [1] * ndim
-        else:  # covers list and array-like inputs
-            self._scale = list(scale)
+            scale = [1] * ndim
         if translate is None:
-            self._translate = [0] * ndim
-        else:  # covers list and array-like inputs
-            self._translate = list(translate)
-        self._scale_view = np.ones(ndim)
-        self._translate_view = np.zeros(ndim)
-        self._translate_grid = np.zeros(ndim)
+            translate = [0] * ndim
+
+        # Create main transform mapping data to a world-like coordinate
+        self._transform = ScaleTranslate(scale, translate)
+        # Create additional transform mapping viewed data to data coordinate
```

Comment:
This comment is still unclear to me, although you went over the outlines of it during group meeting. Based on our discussion in the meeting I see two scenarios where this is useful:

- you are using a pyramid, so the scale of the data at layer 1 (starting at 0) is (say) 2x the scale input by the user. Also, the coordinate of the 0th pixel is shifted slightly. (?) 
- you sent in an image that is bigger than the maximum texture size, so we downsample it 2x before sending it to vispy, so again, transform view will be 2x the scale.

It would be useful to put one of these worked examples in a comment here, just to clarify what "viewed data to data coordinate" means.

Another way of phrasing it that might make more sense to me:

"Additional transform for cases where the displayed data is *not* the base data given by the user. In such cases, we need to transform the coordinates from the displayed data space to the original input data space. One example of such a situation is viewing a higher level of an input pyramid, where `_transform` contains the transform only from the base level to world coordinates."

It's wordy, and might seem like overkill, but I really think it's worth erring on the side of wordiness here. One example I often refer to is the CPython implementation of dictionaries, which has about a 1-1 comment:code ratio. Which is great because I find the code incomprehensible! 😂 

https://github.com/python/cpython/blob/8bae21962bab2fac7630982abd73676b89930902/Objects/dictobject.c#L134-L224

Reply:
comment added, text slightly modified to include even more words - I agree with the sentiment and the practice